### PR TITLE
Add UTs for NOMC controller

### PR DESCRIPTION
**Pull request by bharath-b-rh**
ok  	github.com/openshift/node-observability-operator/pkg/operator/controller/machineconfig	1.020s	coverage: 73.2% of statements
?   	github.com/openshift/node-observability-operator/pkg/operator/controller/machineconfig/machineconfigfakes	[no test files]

Coverage will be increased, by covering the client failure scenarios using the counterfeiter implementation in another PR.

### CODE REVIEW

File: pkg/operator/controller/machineconfig/mcp.go [hunk 1]
```diff
@@ -168,15 +168,6 @@ func (r *MachineConfigReconciler) CheckNodeObservabilityMCPStatus(ctx context.Co
 
 	r.Lock()
 
-	if !r.CtrlConfig.Status.IsDebuggingEnabled() {
```
Comment (bharath-b-rh) [inline pkg/operator/controller/machineconfig/mcp.go:171]: Condition handled [here](https://github.com/openshift/node-observability-operator/commit/9d1ecfca265c761477ff0827b79b145ceff3eb5d), before calling the method, hence removed redundant check.

File: pkg/operator/controller/machineconfig/mcp.go [hunk 2]
```diff
@@ -235,15 +226,6 @@ func (r *MachineConfigReconciler) checkWorkerMCPStatus(ctx context.Context) (ctr
 
 	r.Lock()
 
-	if r.CtrlConfig.Status.IsDebuggingEnabled() && !r.CtrlConfig.Status.IsDebuggingFailed() {
```
Comment (bharath-b-rh) [inline pkg/operator/controller/machineconfig/mcp.go:238]: Condition handled [here](https://github.com/openshift/node-observability-operator/blob/84ffe4e80f367bfd2e03b6073392375b7f0feb1b/pkg/operator/controller/machineconfig/controller.go#L426), while calling the method, hence removed redundant check.

**Review by josefkarasek — APPROVED** [review]
/lgtm

**Review by sherine-k — APPROVED** [review]
Kudos @bharath-b-rh 
I find the use of functions asExpected and preReq very smart.
I was wondering if you think that it's worth it to add a asExpected for the monitorProgress tests in order to check the status we get as output...
It's also a possibility that we relegate this to E2E tests.

**Comment by saschagrunert** [comment]
Came late to that party, greta work @bharath-b-rh! :raised_hands:

**Pull request merged**